crates/oxc_cli/src/codeowners.rs

```diff
@@ -0,0 +1,378 @@
+//! Code from https://github.com/softprops/codeowners
+//!
+//! Codeowners provides interfaces for resolving owners of paths within code
+//! repositories using
+//! Github [CODEOWNERS](https://help.github.com/articles/about-codeowners/)
+//! files
+//!
+//! # Examples
+//!
+//! Typical use involves resolving a CODEOWNERS file, parsing it,
+//! then querying target paths
+//!
+//! ```no_run
+//! extern crate codeowners;
+//! use std::env;
+//!
+//! fn main() {
+//!   if let (Some(owners_file), Some(path)) =
+//!      (env::args().nth(1), env::args().nth(2)) {
+//!      let owners = codeowners::from_path(owners_file);
+//!      match owners.of(&path) {
+//!        None => println!("{} is up for adoption", path),
+//!        Some(owners) => {
+//!           for owner in owners {
+//!             println!("{}", owner);
+//!           }
+//!        }
+//!      }
+//!   }
+//! }
+//! ```
+use glob::Pattern;
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::{
+    fmt,
+    fs::File,
+    io::{BufRead, BufReader, Read},
+    path::Path,
+    str::FromStr,
+};
+
+/// Various types of owners
+///
+/// Owners supports parsing from strings as well as displaying as strings
+///
+/// # Examples
+///
+/// ```rust
+/// let raw = "@org/team";
+/// assert_eq!(
+///   raw.parse::<codeowners::Owner>().unwrap().to_string(),
+///   raw
+/// );
+/// ```
+#[derive(Debug, PartialEq, Eq)]
+pub enum Owner {
```

Comment:
If we own this crate, we can just remove this as we don't discriminate this in our cli, but I left it here because there's no reason to change this code if it works already.

Reply:
It might be worth removing because it takes time and executes regular expressions that we don't actually care about here: https://github.com/web-infra-dev/oxc/pull/1083/files#diff-9bb450403ceef01ed5ba546bd9ab2cafd0147109fbdc217b7aabf7b903a238deR81-R85

Reply:
We can always remove it if it shows up in profile. The regex will be run on small strings so the performance penalty won't be too noticeable.

Reply:
I don’t think it’s worth exploring this now.